n64/src/ap.c

```diff
@@ -1141,13 +1141,75 @@ void ap_update() {
 
 void ap_check_enough_notes(u16 start, u16 end) {
   if (start == end || start > end) return;
+
+  const char* names[] = {
+    "EGG AIM",
+    "BREEGULL BLASTER",
+    "GRIP GRAB",
+    "FIRE EGGS",
+    "BILL DRILL",
+    "BEAK BAYONET",
+    "GRENADE EGGS",
+    "SPLIT UP",
+    "PACK WHACK",
+    "AIRBORNE EGG AIMING",
+    "ICE EGGS",
+    "WING WHACK",
+    "SUB-AQUA EGG AIMING",
+    "TALON TORPEDO",
+    "CLOCKWORK KAZOOIE EGGS",
+    "SPRINGY STEP SHOES",
+    "TAXI PACK",
+    "HATCH",
+    "CLAW CLAMBER BOOTS",
+    "SNOOZE PACK",
+    "LEG SPRING",
+    "SHACK PACK",
+    "GLIDE",
+    "SACK PACK"
+  };
+
+  const int UNLOCK_LIMIT = 5;
+  int unlocks[UNLOCK_LIMIT];
+  int n_unlocks = 0;
+
   for (int i = 0; i < sizeof(ap_memory.pc.settings.silo_requirements)/sizeof(*ap_memory.pc.settings.silo_requirements); i++) {
     u16 amount = ap_memory.pc.settings.silo_requirements[i];
     if (start < amount && end >= amount) {
-      ap.internal_icon = BT_ZOOMBOX_ICON_JAMJARS;
-      strcpy(ap.internal_message, "YOU HAVE ENOUGH NOTES FOR A NEW MOVE!");
-      break;
+      if (n_unlocks < UNLOCK_LIMIT) {
+        unlocks[n_unlocks] = i;
+      }
+      ++n_unlocks;
```

Comment:
```suggestion
      unlocks[n_unlocks++] = i;
      if (n_unlocks == UNLOCK_LIMIT+1) break;
```
This allows exiting the loop early as soon as we've surpassed our limit.

Reply:
added a break clause instead to avoid the extra element